wakuv2/waku.go

```diff
@@ -1023,12 +1023,16 @@ func (w *Waku) Send(pubsubTopic string, msg *pb.WakuMessage) ([]byte, error) {
 	return envelope.Hash().Bytes(), nil
 }
 
-func (w *Waku) query(ctx context.Context, peerID peer.ID, pubsubTopic string, topics []common.TopicType, from uint64, to uint64, requestID []byte, opts []legacy_store.HistoryRequestOption) (*legacy_store.Result, error) {
+func (w *Waku) query(ctx context.Context, peerID peer.ID, pubsubTopic string, topics []common.TopicType, from uint64, to uint64, requestID []byte, cursor *storepb.Index, opts []legacy_store.HistoryRequestOption) (*legacy_store.Result, error) {
```

Comment:
Well I find 9 arguments to be a bit too many 😄

Reply:
hm.. you're right. Probably a good moment to use a struct here :)

Reply:
In the latest commit, I did a small refactor to reduce the number of parameters! pls have a look again :).